src/mscorlib/src/System/Text/StringBuilder.cs

```diff
@@ -999,6 +999,130 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             return this;
         }
 
+        public StringBuilder Append(StringBuilder value)
+        {
+            Contract.Ensures(Contract.Result<StringBuilder>() != null);
+
+            if (value != null && value.Length > 0)
+            {
+                if (value != this)
```

Comment:
@AlexRadch yes -- making all of `StringBuilder` thread safe is not the goal. 

I am worried that we are doing unsafe operations while copying `StringBuilder` data and that the source can change while we are copying. I am concerned that that will lead to random data being copied into the new `StringBuilder`.